docs/apidoc/template_circuits.rst

```diff
@@ -0,0 +1,107 @@
+.. _qiskit-template-circuits:
+
+.. automodule:: qiskit.circuit.library.templates
+   :no-members:
+   :no-inherited-members:
+   :no-special-members:
+
+Template Circuits
+================
```

Comment:
Just a quick comment: the length of the line with "="s should match exactly with the length of the line just above it (the same applies to the lines with "-"s).